project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -346,81 +343,88 @@ public static String getAssignmentsData (String courseID) {
         Query query = new Query("Answer").setFilter(propertyFilter);
-        List<Answer> answers = new ArrayList<>();
-        for (Entity entity : query(query)) {
-            String filePath = (String) entity.getProperty("filePath");
-            String parsedAnswer = (String) entity.getProperty("parsedAnswer");
-            Long scoreLong = (Long) entity.getProperty("score");
-            int score = Math.toIntExact(scoreLong);
-            Key answerKey = entity.getKey();
-            Answer currAnswer = new Answer(filePath, parsedAnswer, score, assignmentKey, questionKey, answerKey);
-            answers.add(currAnswer);
-        }
-        String json = new Gson().toJson(answers);
-        return json;
-    }
-    catch (Exception e) {
-        System.out.println("answers weren't found");
-        e.printStackTrace(System.out);
-        return "error";
-    }
-  }
-
-  public static String createGroups(String assignmentKey, String questionKey) {
-    Filter propertyFilterAssignment = new FilterPredicate("assignmentKey", FilterOperator.EQUAL, assignmentKey);
-    Filter propertyFilterQuestion = new FilterPredicate("questionKey", FilterOperator.EQUAL, questionKey);
-    CompositeFilter propertyFilter = CompositeFilterOperator.and(propertyFilterAssignment, propertyFilterQuestion);
-    Query query = new Query("Answer").setFilter(propertyFilter);
-    HashMap<String, ArrayList<Entity>> parsedAnswerGroups = new HashMap<String, ArrayList<Entity>>();
-    // loop through answers and see if parsed answer already exists. if so, add answer entity to arraylist. if not exists, init new array and add
-    for (Entity answer : query(query)) {
-        String parsedAnswer = (String) answer.getProperty("parsedAnswer");
-        if (parsedAnswerGroups.get(parsedAnswer) == null) {
-            ArrayList<Entity> temp = new ArrayList<Entity>();
-            temp.add(answer);
-            parsedAnswerGroups.put(parsedAnswer, temp);
-        }
-        else {
-            parsedAnswerGroups.get(parsedAnswer).add(answer);
-        }
-    }
-    HashMap<Group, ArrayList<Answer>> groups = new HashMap<Group, ArrayList<Answer>>();
-    for (HashMap.Entry<String, ArrayList<Entity>> entry : parsedAnswerGroups.entrySet()) {
-        Entity groupEntity = addGroup((int) 0, questionKey);
-        int score = (int) groupEntity.getProperty("score");
-        Key groupKey = groupEntity.getKey();
-        Group group = new Group(score, questionKey, groupKey);
-        ArrayList<Answer> answerList = new ArrayList<Answer>();
-        // created group entity and class @ this point
-        // now we loop through all the answers and map them to a group
-        for (Entity answer : entry.getValue()) {
-            String filePath = (String) answer.getProperty("filePath");
+        HashMap<String, ArrayList<Entity>> parsedAnswerGroups = new HashMap<String, ArrayList<Entity>>();
```

Comment:
Declare variables with their interface type and use the diamond syntax: `Map<String, List<Entity>> parsedAnswerGroups = new HashMap<>();`